lib/client/api.go

```diff
@@ -798,7 +798,12 @@ func (tc *TeleportClient) getTargetNodes(ctx context.Context, proxy *ProxyClient
 		}
 	}
 	if len(nodes) == 0 {
-		retval = append(retval, net.JoinHostPort(tc.Host, strconv.Itoa(tc.HostPort)))
+		// Precent code the host:port. This is done so the returned address can be
+		// passed to url.Parse where the host:port is placed within the userinfo
+		// part of the URL which only allows alpha numberic plus some punctuation.
+		// See the following for details: https://github.com/golang/go/issues/23392.
+		escaped := url.QueryEscape(net.JoinHostPort(tc.Host, strconv.Itoa(tc.HostPort)))
```

Comment:
I would like to encapsulate this logic in the ssh subsystem call library that already deals with parsing `@` instead of sprinkling this over the place. Also is escaping is really necessary? can we detect IPV6 address as is?

Reply:
The issue isn't the SSH subsystem, but that we pack an IPv6 address into the userinfo of a URL then try and call `url.Parse()` on it. The userinfo part of a URL can't contain `[` and `]`, see the following ticket: https://github.com/golang/go/issues/23392.

The other approach we can take is do some refactoring around how we store node/cluster info in a `utils.NetAddr`, however that seemed a more invasive fix.

Reply:
Still not sure I understand, where do we treat this address as URL and why do we do it?